File: 36-valid-sudoku/valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {

        for(int i = 0; i < 9; i++){
            vector<int>check(10, 0);
            for(int j = 0; j < 9; j++){
                if(board[i][j] == '.'){
                    continue;
                }
                else{
                    int x = board[i][j] - '0';
                    if(check[x] > 0) return false;
                    check[x] = 1;
                }
            }
            check.clear();
        }
        for(int j = 0; j < 9; j++){
            vector<int> check(10,0);
            for(int i = 0; i < 9; i++){
                if(board[i][j] == '.'){
                    continue;
                }
                else{
                    int x = board[i][j] - '0';
                    if(check[x] > 0) return false;
                    check[x] = 1;
                }
            }
            check.clear();
        }

        for(int row=0; row<9; row+=3)
        {
            for(int col=0; col<9; col+=3)
            {
                vector<int> check(10,0);

                for(int i=row;i<row+3;i++)
                {
                    for(int j=col;j<col+3;j++)
                    {
                        if(board[i][j] == '.'){
                            continue;
                        }
                        else{
                            int x = board[i][j] - '0';
                            if(check[x] > 0) return false;
                            check[x] = 1;
                        }
                    }
                }
                check.clear();
            }
        }
        return true;
    }
};

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:

        n = len(board)
        m = len(board[0])

        for i in range(n):
            check = []
            for x in board[i]:
                if x == ".":
                    continue
                if x in check:
                    return False
                else:
                    check.append(x)
        
        for i in range(n):
            check = []
            for j in range(n):
                if board[j][i] == ".":
                    continue
                if board[j][i] in check:
                    return False
                else:
                    check.append(board[j][i])
        for i in range(0,9,3):
            for j in range(0,9,3):
                check = []
                for k in range(i, i+3):
                    for p in range(j, j+3):
                        if board[k][p] == ".":
                            continue
                        if board[k][p] in check:
                            return False
                        else:
                            check.append(board[k][p])
        
        return True
                         